src/com/codecool/HTTPHandler.java

```
@@ -0,0 +1,71 @@
+package com.codecool;
+
+import org.jsoup.Connection;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+
+public class HTTPHandler {
+    private static final String USER_AGENT =
+            "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/535.1 (KHTML, like Gecko) Chrome/13.0.782.112 Safari/535.1";
+    private List<String> links = new LinkedList<String>();
+    private Document htmlDocument;
+
+    public boolean crawl(String url) {
+        try {
+            Connection connection = Jsoup.connect(url).userAgent(USER_AGENT);
+            Document htmlDocument = connection.get();
+            this.htmlDocument = htmlDocument;
+
+            if(connection.response().statusCode() == 200){ // 200 is the HTTP OK status code
+                System.out.println("\n**Visiting** Received web page at " + url);
+            }
+
+            if(!connection.response().contentType().contains("text/html")) {
+                System.out.println("**Failure** Retrieved something other than HTML");
+                return false;
+            }
+
+            Elements linksOnPage = htmlDocument.select("a[href]");
+            System.out.println("Found (" + linksOnPage.size() + ") links");
+
+            for(Element link : linksOnPage) {
+                this.links.add(link.absUrl("href"));
+            }
+            return true;
+        } catch(IOException ioe) {
```

1. It's better not to make the `try` block too broad. How about wrapping only the code that you expect to fail? (that is, only `connect() and connection.get()` calls, I think).

2. When you catch an exception, at least some minimal logging would be good instead of swallowing it silently; for instance "Failed to retrieve <url>: <exception name>".